lib/travis/build/addons/pkg.rb

```diff
@@ -0,0 +1,69 @@
+require 'travis/build/addons/base'
+require 'shellwords'
+
+module Travis
+  module Build
+    class Addons
+      class Pkg < Base
+        SUPPORTED_OPERATING_SYSTEMS = %w[
+          freebsd
+        ].freeze
+
+        def before_prepare?
+          SUPPORTED_OPERATING_SYSTEMS.any? do |os_match|
+            data[:config][:os].to_s == os_match
+          end
+        end
+
+        def before_prepare
+          return if config_pkg.empty?
+          sh.newline
+          sh.fold('pkg') do
+            install_pkg
+          end
+          sh.newline
+        end
+
+        def before_configure?
+          # keeping empty for now
+        end
+
+        def before_configure
+          sh.echo "Configuring default pkg options", ansi: :yellow
+          tmp_dest = "${TRAVIS_TMPDIR}/99-travis-pkg-conf"
+          sh.file tmp_dest, <<~PKG_CONF
+            ASSUME_ALWAYS_YES=YES
+            FETCH_RETRY=5
+            FETCH_TIMEOUT=30
+          PKG_CONF
+          sh.cmd %Q{su -m root -c "mv #{tmp_test} ${TRAVIS_ROOT}/usr/local/etc/pkg.conf"}
```

Comment:
What's the expansion of `${TRAVIS_ROOT}`? I think this path is expected to just be `/usr/local/etc/pkg.conf` (see [here](https://www.freebsd.org/cgi/man.cgi?query=pkg.conf&sektion=5&apropos=0&manpath=FreeBSD+12.1-RELEASE+and+Ports)) so unless it's empty, this could perhaps be causing it to ignore the `ASSUME_ALWAYS_YES` option.